minkindr/include/kindr/minimal/implementation/quat-sim-transform-inl.h

```diff
@@ -17,12 +17,46 @@ QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate(
 }
 
 template <typename Scalar>
+QuatSimTransformTemplate<Scalar>::QuatSimTransformTemplate(
+    const Vector7& log_vector)
+    : T_A_B_(Eigen::Matrix<Scalar, 6, 1>(log_vector.template head<6>())),
+      scale_A_B_(log_vector(6)) {
+  CHECK_GT(scale_A_B_, 0.);
+}
+
+template <typename Scalar>
+typename QuatSimTransformTemplate<Scalar>::Vector
+QuatSimTransformTemplate<Scalar>::operator*(const Vector& rhs) const {
+  return T_A_B_ * (scale_A_B_ * rhs);
+}
+
+template <typename Scalar>
 typename QuatSimTransformTemplate<Scalar>::Vectors
 QuatSimTransformTemplate<Scalar>::operator*(const Vectors& rhs) const {
   return T_A_B_.transformVectorized(scale_A_B_ * rhs);
 }
 
 template <typename Scalar>
+QuatSimTransformTemplate<Scalar> QuatSimTransformTemplate<Scalar>::operator*(
+    const QuatSimTransformTemplate<Scalar>& rhs) const {
+  // Rl*sl tl         Rr*sr tr         Rl*sl*Rr*sr Rl*sl*tr+tl
+  // 0     1     *    0     1     =    0           1
+  //
+  // -> R = Rl*Rr;  t = Rl*sl*tr+tl = Siml*tr;  s = sl*sr
+  return QuatSimTransformTemplate<Scalar>(
+      QuatTransformationTemplate<Scalar>(
+          T_A_B_.getRotation() * rhs.T_A_B_.getRotation(),
+          operator *(rhs.T_A_B_.getPosition())),
```

Comment:
would it be possible to explictly write out the operation, e.g. ``*this * rhs.T_A_B_.getPos()`` or something like that?